### ISSUE TITLE
Add a compatability to drop item created by other plugins

### ISSUE THREAD

**Issue by mizuo-san** [thetealviper/Spigot_FarmCraft#6]
As an option, I suggest adding plugin compatibility with others like UltimateFood or similar so that it is possible to combine two plugins

**Comment by thetealviper** [comment]
It should already be possible to do this. Simply configure the items similarly across the two plugins, and if you happened to produce the same item with both then they should automatically cooperate. If they do not, there is a tiny chance that the other plugins are using a special NBT tag when they create the items for identification. If they do this, there is nothing I can do to make a similar item because using that special tag is specific to the plugin that makes it. I can not pretend to be a plugin which I am not.

tldr; Either it's already possible, or if it's not then it's because of an impossible thing to achieve.

**Comment by mizuo-san** [comment]
Main problem that plugin use custom data
![image](https://github.com/user-attachments/assets/cf60ce7e-6277-4806-ad99-1cfa5099806d)
I dont know how to change custom data for item that i wanna get from crop while using ur plugin.

**Comment by thetealviper** [comment]
Hey Mizuo. After reading guides on the features this would require, they suggested that what you wanted was not possible. After looking at the actual Spigot documentation and code though, I believed it was possible, so I attempted to implement it. I believe I succeeded. I got it to work, but you know the plugins that aren't mine better and you'd know if it works properly better, so if you could help me test I would appreciate it. I'm attaching the test build in a zip file, and explaining the files I used to give an example of how to set things up. For context, I'm using UltimateFood as a test.

.ZIP Test Build: 
[FarmCraftTestBuild.zip](https://github.com/user-attachments/files/17091408/FarmCraftTestBuild.zip)

--------------------
Step 1. UltimateFood item config. For UltimateFood, I decided to use the Coca Cola default item. The yml for this looks like below. The important parts that would need to be copied by my plugin item configuration are displayname, description (lore), foodmaterial ("id" in my plugin), and custommodeldata. Everything else only matters behind the scenes for UltimateFood and isn't stored on the item itself. EXCEPT FOR "uniqueID". Make a note of this because this is what will be added to the custom data later.
```yml
{
  "uniqueID": "Coca_Cola",
  "displayName": "&fCoca Cola",
  "description": [],
  "instantEat": false,
  "foodMaterial": "APPLE",
  "headTexture": "",
  "customModelData": 30094,
  "permission": "",
  "price": 0.0,
  "saturation": 0.13661216,
  "hunger": 0.1275,
  "health": 0.0,
  "commands": [],
  "recipe": {
    "shape": [
      "aaa",
      "aaa",
      "aaa"
    ],
    "materials": {
      "a": "AIR"
    },
    "craft_amount": 1
  }
}
```
-------------------------
Step 2. FarmCraft item config. My FarmCraft item config yml can be seen below. The important things to carry over directly from the UltimateFood config are id (material), name (displayname), and custommodeldata tag. Finally, it's not specified in the UltimateFood config but to make an identical item to the one that plugin produces, you would add "custompersistentdata" exactly how I have it where the last bit is equal to whatever you have "uniqueID" set to in the UltimateFood config.
```yml
Required_Light: -1
# Set to -1 if no required light
Require_Water_Radius: -1
# Set to -1 if no required water
Required_XP: -1
# Set to -1 if no required xp
Grow_Time: 10
# This is in seconds

# headtexture values can be found at "https://minecraft-heads.com"
# 1. Find a head you want
# 2. Click on it
# 3. The value you want to copy and paste in this config is "Minecraft-URL"

Seed:
  id: APPLE
  name: '&fCoca Cola'
  tags:
   - "custommodeldata:30094"
  custompersistentdata:
   - "ultimatefood:modifiers:Coca_Cola"

Harvests:
  1:
    percent: 0
    chance: 50
    headtexture: 59970d085bf5aae875165f8337896242c12cd11299df89891a8a813adeb8
    xp: 0
    FirstItemToDrop:
      id: STICK
      amount: 1
```

**Comment by thetealviper** [comment]
Actually I'm pretty confident it works and I'm just going to push the update because I'm impatient 😭

**Comment by mizuo-san** [comment]
It works great, thank you!
